__tests__/api.test.js

```diff
@@ -330,3 +330,30 @@ describe("GET api/users", () => {
     });
   });
 });
+
+describe("GET /api/articles with topic query", () => {
+  it("should return an array of article with only the topic specified in the path", async () => {
+    const { body } = await request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200);
+    body.articles.forEach((article) => {
+      expect(article).toMatchObject({
+        author: expect.any(String),
+        title: expect.any(String),
+        article_id: expect.any(Number),
+        topic: "mitch",
+        created_at: expect.any(String),
+        votes: expect.any(Number),
+        article_img_url: expect.any(String),
+        comment_count: expect.any(String),
+      });
+    });
+  });
+
+  it("should return an error if an invalid topic was input", async () => {
+    const { body } = await request(app)
+      .get("/api/articles?topic=notatopic")
+      .expect(400);
+    expect(body.message).toBe("Invalid Topic");
+  });
```

Comment:
Couldn't 'notatopic' _technically_ be a topic at some point? Would be a silly topic but could happen I guess!

Another test I would suggest is another 200 when the topic exists but has not articles :)